spectec/src/il2al/il2il.ml

```diff
@@ -209,12 +209,12 @@ let rec collect_unified template e = if eq_exp template e then [], [] else
     | ListE es1, ListE es2 ->
         List.fold_left2 (fun acc e1 e2 -> pairwise_concat acc (collect_unified e1 e2)) ([], []) es1 es2
     | CallE (_, as1), CallE (_, as2) -> collect_unified_args as1 as2
-    | _ -> failwith "Impossible collect_unified"
+    | _ -> Util.Error.error template.at "il2il" "cannot match template with expression"
```

Comment:
Indeed. Instead of "template", I used "unify" to express the process. I also changed the line information to that of the expression to which we are applying the template.